messagingpush/src/main/java/io/customer/messagingpush/CustomerIOFirebaseMessagingReceiver.kt

```diff
@@ -0,0 +1,26 @@
+package io.customer.messagingpush
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import io.customer.messagingpush.di.moduleConfig
+import io.customer.messagingpush.di.pushTrackingUtil
+import io.customer.messagingpush.extensions.getSDKInstanceOrNull
+import io.customer.sdk.data.request.MetricEvent
+
+class CustomerIOFirebaseMessagingReceiver : BroadcastReceiver() {
```

Comment:
At this time, this class's only responsibility is delivered push metrics. I think we should add a comment to this class explaining this limited responsibility and why for future readers. 

I am comfortable with relying on this broadcast receiver more in the future once we test it in production and see it's reliable. But until then, keeping it limited adds trust to our code.

Reply:
Added comments to it

Reply:
I have a separate comment from the one above. 

Why does the class name contain `Firebase` but the comment and the code indicates that this is a class only used by GCM? I am assuming this was a typo?